### PR TITLE
Cleanup Playback Tests

### PR DESCRIPTION
**Pull request by alzimmermsft**
Disables most buffered upload tests as they use variables that cannot be captured by the playback framework.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy [hunk 1]
```diff
@@ -189,7 +189,7 @@ class APISpec extends Specification {
     }
 
     static boolean liveMode() {
-        return setupTestMode() == TestMode.RECORD
+        return setupTestMode() == TestMode.RECORD && false
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy]: What does this mean?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobAPITest.groovy [hunk 2]
```diff
@@ -732,7 +750,9 @@ class BlockBlobAPITest extends APISpec {
         thrown(StorageException)
     }
 
+    // Only run these tests in live mode as they use variables that can't be captured.
     @Unroll
+    @Requires({ APISpec.liveMode() })
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobAPITest.groovy]: Can we split two two different tests? There are bunch of small data which we are able to test in playback.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy]: I was forcing the live test only tests not to record playback records while I was updating the other tests, removed.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobAPITest.groovy (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobAPITest.groovy]: The issue isn't the data size but the fact that there are variables that get used inside of the method call that can't be captured by the playback system.

Comment (sima-zhu) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobAPITest.groovy]: Why we cannot record the variables?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy (hunk 1 above)
Comment (vhvb1989) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy]: it looks like a bad hardcoding isn't it?  why bother to call `setupTestMode()` if we are returning false any how
why not just returning false?

Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/APISpec.groovy]: I was forcing the live test only tests not to record playback records while I was updating the other tests, removed.

**Pull request merged**